Src/Notion.Client/RestClient/RestClient.cs

```diff
@@ -174,17 +172,13 @@ private static void AddHeaders(HttpRequestMessage request, IDictionary<string, s
             }
         }
 
-        private void EnsureHttpClient()
+        private HttpClient GetHttpClient()
         {
-            if (_httpClient != null)
```

Comment:
@HermanSchoenfeld This will create a new instance of HttpClient for each request. HttpClient instances are meant to be long-lived, and reused throughout the lifetime of the application